pkg/registry/core/service/storage/rest.go

```diff
@@ -153,7 +174,14 @@ func (rs *REST) NewList() runtime.Object {
 }
 
 func (rs *REST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
-	return rs.services.Get(ctx, name, options)
+	obj, err := rs.services.Get(ctx, name, options)
+	if err != nil {
+		return obj, err
+	}
+	// default on read for pre-existing services
```

Comment:
just for learning purposes, why is this needed now?

Reply:
There are two new fields that has to be defaulted according to:
1- Service configuration (e.g. is it IPv6 or IPv4?)
2- Cluster Default (if it does not have ClusterIP then it falls under cluster defaults).

These things can not be executed in defaulting logic, because that part of api-server does not know how the cluster is configured (it does not have access to configuration flags).